File: content-solutions/queue-deque/queue-recent-visits.java
import java.util.*;

class Solution {
    public List<Integer> solution(int[] times) {
        Queue<Integer> window = new ArrayDeque<>();
        List<Integer> answer = new ArrayList<>();
        for (int t : times) {
            window.offer(t);
            while (window.peek() < t - 3000) window.poll();
            answer.add(window.size());
        }
        return answer;
    }
}

File: content-solutions/queue-deque/queue-recent-visits.js
function solution(times) {
  const window = [];
  let head = 0;
  const answer = [];
  for (const t of times) {
    window.push(t);
    while (window[head] < t - 3000) head += 1;
    answer.push(window.length - head);
  }
  return answer;
}
